### PR TITLE
WD-8729 - feat: Display errors in Config Panel

### PR DESCRIPTION
**Pull request by vladimir-cucu**
## Done

- Displayed get config and submit to juju errors in Config Panel.

## Details

https://warthogs.atlassian.net/browse/WD-8729

### CODE REVIEW

File: src/panels/ConfigPanel/ConfigPanel.tsx [hunk 1]
```diff
@@ -69,6 +75,23 @@ export default function ConfigPanel(): JSX.Element {
   const [savingConfig, setSavingConfig] = useState<boolean>(false);
   const [confirmType, setConfirmType] = useState<ConfirmTypes>(null);
   const [formErrors, setFormErrors] = useState<string[] | null>(null);
+  const [inlineErrors, setInlineError] = useInlineErrors({
```
Comment (huwshimi) [inline src/panels/ConfigPanel/ConfigPanel.tsx:78]: I think we should merge these two error states. To do this we would need to update `useInlineErrors` to output an error for each item if the error is an array, something like this:

```
const errors = inlineErrors.reduce<ReactNode[]>((nodes, { key, error }) => {
    const errorItems = mapping && key in mapping ? mapping[key](error) : error;
    const filteredItems = (Array.isArray(errorItems) ? errorItems : [errorItems]).filter(Boolean);
    return nodes.concat(filteredItems);
  }, []);
```

Comment (vladimir-cucu) [reply src/panels/ConfigPanel/ConfigPanel.tsx]: Done!

File: src/panels/ConfigPanel/ConfigPanel.tsx [hunk 2]
```diff
@@ -194,7 +222,7 @@ export default function ConfigPanel(): JSX.Element {
     setEnableSave(false);
     setConfirmType(null);
     if (errors?.length) {
-      setFormErrors(errors);
+      setInlineError(InlineErrors.FORM, errors);
       return;
     }
     await getConfig(
```
Comment (huwshimi) [inline src/panels/ConfigPanel/ConfigPanel.tsx:228]: I might be missing something, but I can't see why it fetches the config once it saves successfully as all it does after that is close the panel.

If we remove this call then `getConfig` no longer needs to return a promise, so we can avoid the `void` inside `getConfigCallback`.

**Review by huwshimi — APPROVED** [review]
This looks good, I've just made one suggestion that's not something you've changed but should improve performance and avoid a `void`.

**Pull request merged**